packages/core/spec/node/asciidoctor.spec.js

```diff
@@ -1723,6 +1723,82 @@ header_attribute::foo[bar]`
       const result = asciidoctor.convert('content', options)
       expect(result).to.contain('<dummy>content</dummy>')
     })
+    it('should retrieve backend traits from a converter class', () => {
+      class TEIConverter {
+        constructor (backend, _) {
+          this.backend = backend
+          this.backendTraits = {
+            basebackend: 'xml',
+            outfilesuffix: '.xml',
+            filetype: 'xml',
+            htmlsyntax: 'xml'
+          }
+          this.transforms = {
+            embedded: (node) => {
+              return `<tei>${node.getContent()}</tei>`
+            }
+          }
+        }
+
+        convert (node, transform) {
+          const name = transform || node.node_name
+          if (name === 'paragraph') {
+            return this.convertParagraph(node)
+          }
+          return this.transforms[name](node)
+        }
+
+        convertParagraph (node) {
+          return node.getContent()
+        }
+      }
+
+      asciidoctor.ConverterFactory.register(TEIConverter, ['tei'])
```

Comment:
Here we use a class. The processor will call the constructor with two arguments, the backend name and a list of options.